compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java

```diff
@@ -4175,19 +4175,23 @@ private BLangMatchPattern transformMatchPattern(Node matchPattern) {
                     (BLangListMatchPattern) TreeBuilder.createListMatchPattern();
             bLangListMatchPattern.pos = matchPatternPos;
 
-            for (Node memberMatchPattern : listMatchPatternNode.matchPatterns()) {
-                BLangMatchPattern bLangMemberMatchPattern = transformMatchPattern(memberMatchPattern);
+            SeparatedNodeList<Node> matchPatterns = listMatchPatternNode.matchPatterns();
+            int matchPatternListSize = matchPatterns.size();
+            for (int i = 0; i < matchPatternListSize - 1; i++) {
```

Comment:
Here, we have a bug. When there are no members in the `matchPatterns`, this may give `IndexOutOfBoundsException` for `index = -1`. Better, if we can early-return if `matchPatternListSize < 1`

Reply:
Thanx will update